opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java

```diff
@@ -861,6 +862,16 @@ private static void splitByComma(String value, List<String> result) {
         }
     }
 
+    /**
+     * Add cookie header to the response along with extra attributes.
+     * @param response servlet response
+     * @param name name of the cookie
+     * @param value value of the cookie
+     */
+    public static void addCookie(HttpServletResponse response, String name, String value) {
+        response.addHeader("Set-cookie", name + "=" + value + "; SameSite=Strict; Secure");
```

Comment:
Even though I did not see repeating headers in the response maybe this should just record the cookie in a map and the actual headers will be added in a filter.

Reply:
found a solution to this in https://stackoverflow.com/a/51576089/11582827